packages/kyc-oracle/KycOracle.js

```diff
@@ -0,0 +1,231 @@
+const axios = require("axios");
+const ethers = require("ethers");
+const express = require("express");
+const etherpass = require("etherpass");
+
+const { getChallenge } = etherpass;
+const { verifyEthSignature } = etherpass;
+const sqlite3 = require("sqlite3");
+const sqlite = require("sqlite");
+const bodyParser = require("body-parser");
+
+class KycOracle {
+  /**
+   * Constructor for KycOracle
+   * @param {string} privateKey              The private key of the address that has the administration permission.
+   *                                           If used, `adminAddress` is not needed and will be derived.
+   * @param {string} adminAddress             The address that has the administration permission to edit the whitelist.
+   * @param {string} whitelistAddress        The address the whitelist.
+   * @param {Object} loader                  The loader for loading the contract interfaces. Usually a TruffleLoader.
+   * @param {Object} provider                Ethers provider that allows access to an ethereum node.
+   * @param {Number} [port]                  The port the oracle will serve on
+   */
+  constructor({ privateKey, adminAddress, apiKey, loader, provider, dbPath, port = 3000 }) {
+    if (privateKey) {
+      this.wallet = new ethers.Wallet(privateKey, this.provider);
+      this.adminAddress = this.wallet.address;
+    } else {
+      this.wallet = provider.getSigner(adminAddress);
+      this.adminAddress = adminAddress;
+    }
+    console.log("Transactions will be signed from ", this.adminAddress);
+
+    this.apiKey = apiKey;
+    this.dbPath = dbPath;
+
+    if (!this.dbPath) {
+      this.dbPath = "kycSessions.db";
+    }
+
+    this.loader = loader;
+    this.provider = provider;
+
+    this.app = express();
+
+    this.app.use(function (req, res, next) {
+      const regex = /(.*colony\.io)/;
+      const origin = req.get("origin");
+      const matches = regex.exec(origin);
+      if (matches) {
+        res.header("Access-Control-Allow-Origin", matches[1]);
+      }
+      next();
+    });
+
+    this.app.use(bodyParser.json());
+
+    this.app.get("", async (req, res) => {
+      return res.status(200).send("KycOracle is running!");
+    });
+
+    this.app.post("/auth/challenge", (req, res) => {
+      try {
+        const { address } = req.body;
+        const challenge = getChallenge(address);
+        return res.json({ challenge });
+      } catch (err) {
+        return res.status(500).send(err);
+      }
+    });
+
+    this.app.post("/auth/token", async (req, res) => {
+      try {
+        if (typeof req.body.challenge !== "string" || typeof req.body.signature !== "string") {
+          throw new Error("Invalid challenge/signature");
+        }
+
+        // This throws if bad signature
+        const address = verifyEthSignature(req.body.challenge, req.body.signature);
```

Comment:
Why not just put this line in the try/catch?

Reply:
I don't follow? It's already inside one?

Reply:
Why check the arguments and throw an exception in the previous lines?